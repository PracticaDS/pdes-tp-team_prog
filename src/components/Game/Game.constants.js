/* eslint-disable no-use-before-define */
import crafter from '../../assets/crafter.png'
import furnace from '../../assets/furnace.png'
import seller from '../../assets/seller.png'
import starter from '../../assets/starter.png'
import transporter from '../../assets/transporter.png'
// Material images
import gold from '../../assets/gold.png'
import iron from '../../assets/iron.png'
import silver from '../../assets/silver.png'
import copper from '../../assets/copper.png'
import bronze from '../../assets/bronze.png'
import zinc from '../../assets/zinc.png'
import whiteGold from '../../assets/whiteGold.png'
import chineseSilver from '../../assets/chineseSilver.png'
//
import deleteOption from '../../assets/delete.png'
import rotateOption from '../../assets/rotate.png'
import moveOption from '../../assets/move.png'

import { Starter, Seller, Crafter, Furnace, Transporter } from '../../utils/machineUtils'
import { DOWN } from '../../utils/directions'

export const starterMachine = {
  id: 1,
  image: starter,
  name: 'Starter',
  type: Starter,
  buy: 800,
  sell: 400,
  frequency: 1,
  direction: DOWN,
  metadata: {
    selectedMaterial: null,
    availableMaterials: [],
  },
  process(materials, func) {
    func(this.direction, [], [{ ...this.metadata.selectedMaterial, frequency: this.frequency }])
  },
}

export const sellerMachine = {
  id: 2,
  image: seller,
  name: 'Seller',
  type: Seller,
  buy: 1000,
  sell: 500,
  frequency: 1,
  direction: DOWN,
  process: (materials, func) => {
    func(null, materials, { name: 'currency' })
  },
}

export const crafterMachine = {
  id: 3,
  image: crafter,
  name: 'Crafter',
  type: Crafter,
  buy: 800,
  sell: 400,
  frequency: 1,
  direction: DOWN,
  metadata: {
    recipeSelected: {
      materialsRequired: [{ name: 'harina', quantity: 1 }, { name: 'agua', quantity: 2 }],
      product: {
        name: 'pan',
      },
    },
    availableRecipes: [{ required: ['harina', 'agua'], product: 'pan' }],
  },
  process: (materials, func) => {
    func(this.direction, this.metadata.recipeSelected.materialsRequired, [
      { ...this.metadata.recipeSelected.product, quantity: this.frequency },
    ])
  },
}

export const furnaceMachine = {
  id: 4,
  image: furnace,
  name: 'Furnace',
  type: Furnace,
  buy: 800,
  sell: 400,
  frequency: 1,
  direction: DOWN,
  metadata: {
    recipeSelected: {
      materialsRequired: [{ name: 'gold', quantity: 1 }],
      product: {
        name: 'liquid_gold',
      },
    },
    availableRecipes: [
      { materialsRequired: [{ name: 'gold', quantity: 1 }], product: 'liquid_gold' },
      { materialsRequired: [{ name: 'silver', quantity: 1 }], product: 'liquid_silver' },
    ],
  },
  process: (materials, func) => {
    // los materiales estan incluidos en los requeridos de la receta
    func(this.direction, this.metadata.recipeSelected.materialsRequired, [
      { ...this.metadata.recipeSelected.product, quantity: this.frequency },
    ])
  },
}

export const transporterMachine = {
  id: 5,
  image: transporter,
  name: 'Transporter',
  type: Transporter,
  buy: 800,
  sell: 400,
  frequency: 1,
  direction: DOWN,
  process(materials, func) {
    func(this.direction, materials, materials)
  },
}

export const editionOptions = [
  {
    image: deleteOption,
    title: 'Delete',
  },
  {
    image: rotateOption,
    title: 'Rotate',
  },
  {
    image: moveOption,
    title: 'Move',
  },
]

const goldMaterial = {
  name: 'Gold',
  image: gold,
  price: 20,
}

const silverMaterial = {
  name: 'Silver',
  image: silver,
  price: 10,
}

const ironMaterial = {
  name: 'Iron',
  image: iron,
  price: 5,
}

const zincMaterial = {
  name: 'Zinc',
  image: zinc,
  price: 5,
}

const copperMaterial = {
  name: 'Copper',
  image: copper,
  price: 5,
}

// //////

const bronzeMaterial = {
  name: 'Bronze',
  image: bronze,
  price: 100,
}

const whiteGoldMaterial = {
  name: 'White Gold',
  image: whiteGold,
  price: 100,
}

const chineseSilverMaterial = {
  name: 'Chinese Silver',
  image: chineseSilver,
  price: 100,
}

const whiteGoldRecipe = {
  name: 'White Gold Recipe',
  necessaryMaterials: [
    {
      material: goldMaterial,
      count: 2,
    },
    {
      material: silverMaterial,
      count: 1,
    },
    {
      material: copperMaterial,
      count: 1,
    },
  ],
  currencyCost: 0,
  result: {
    material: whiteGoldMaterial,
    count: 1,
  },
}

const chineseSilverRecipe = {
  name: 'Chinese Silver Recipe',
  necessaryMaterials: [
    {
      material: copperMaterial,
      count: 2,
    },
    {
      material: zincMaterial,
      count: 2,
    },
  ],
  currencyCost: 0,
  result: {
    material: chineseSilverMaterial,
    count: 1,
  },
}

const bronzeRecipe = {
  name: 'Bronze Recipe',
  necessaryMaterials: [
    {
      material: copperMaterial,
      count: 1,
    },
    {
      material: zincMaterial,
      count: 1,
    },
  ],
  currencyCost: 0,
  result: {
    material: bronzeMaterial,
    count: 2,
  },
}

export const materials = [goldMaterial, silverMaterial, zincMaterial, ironMaterial, copperMaterial]

export const recipes = [bronzeRecipe, chineseSilverRecipe, whiteGoldRecipe]

export const areEquals = (machine1, machine2) => machine1 && machine2 && machine1.id === machine2.id
