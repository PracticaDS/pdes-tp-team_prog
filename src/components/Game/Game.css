.gameMainContainer {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background: #243646;
}

.HorizontalGameParts {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.gamePanel {
  padding: 1rem;
}

.gameLeftPanel {
  flex-basis: 10rem;
}

.gameMidPanel {
}

.gameRightPanel {
  flex-basis: 10rem;
}
