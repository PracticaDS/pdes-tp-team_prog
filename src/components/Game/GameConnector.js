import { connect } from 'react-redux'
import { playGame } from '../../actions/GameState'
import { restartCurrency } from '../../actions/Currency'
import { machineSelected } from '../../selectors/GameState'

const connector = Game => {
  const mapStateToProps = state => ({
    machineSelected: machineSelected(state),
  })
  const mapDispatchToProps = dispatch => ({
    restartCurrency: () => dispatch(restartCurrency()),
    startGame: () => dispatch(playGame()),
  })

  return connect(
    mapStateToProps,
    mapDispatchToProps,
  )(Game)
}

export default connector
