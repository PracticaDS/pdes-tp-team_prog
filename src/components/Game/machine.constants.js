import crafter from '../../assets/machines/crafter.png'
import furnace from '../../assets/machines/furnace.png'
import seller from '../../assets/machines/seller.png'
import starter from '../../assets/machines/starter.png'
import transporter from '../../assets/machines/transporter.png'

import { Starter, Seller, Crafter, Furnace, Transporter } from '../../utils/machineUtils'
import { DOWN } from '../../utils/directions'

export const starterMachine = {
  id: 1,
  image: starter,
  name: 'Starter',
  type: Starter,
  price: 800,
  frequency: 1,
  direction: DOWN,
  metadata: {
    selectedMaterial: null,
    availableMaterials: [],
  },
}

export const sellerMachine = {
  id: 2,
  image: seller,
  name: 'Seller',
  type: Seller,
  price: 1000,
  frequency: 1,
  direction: DOWN,
}

export const crafterMachine = {
  id: 3,
  image: crafter,
  name: 'Crafter',
  type: Crafter,
  price: 800,
  frequency: 1,
  direction: DOWN,
  metadata: {
    recipeSelected: null,
    availableRecipes: [],
  },
}

export const furnaceMachine = {
  id: 4,
  image: furnace,
  name: 'Furnace',
  type: Furnace,
  price: 800,
  frequency: 1,
  direction: DOWN,
}

export const transporterMachine = {
  id: 5,
  image: transporter,
  name: 'Transporter',
  type: Transporter,
  price: 800,
  frequency: 1,
  direction: DOWN,
}
