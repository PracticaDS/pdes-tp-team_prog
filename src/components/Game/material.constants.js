import gold from '../../assets/materials/gold.png'
import gold_melted from '../../assets/melted_materials/gold_melted.png'
import iron from '../../assets/materials/iron.png'
import silver from '../../assets/materials/silver.png'
import copper from '../../assets/materials/copper.png'
import bronze from '../../assets/materials/bronze.png'
import zinc from '../../assets/materials/zinc.png'
import whiteGold from '../../assets/materials/whiteGold.png'
import chineseSilver from '../../assets/materials/chineseSilver.png'
import billon from '../../assets/materials/billon.png'

export const materialsImg = {
  Gold: gold,
  Silver: silver,
  Iron: iron,
  Zinc: zinc,
  Copper: copper,
  Bronze: bronze,
  'White Gold': whiteGold,
  'Chinese Silver': chineseSilver,
  Billon: billon,
  DEFAULT: silver,
  'Gold Melted': gold_melted,
  'Silver Melted': silver,
  'Iron Melted': iron,
  'Zinc Melted': zinc,
  'Copper Melted': copper,
  'Bronze Melted': bronze,
  'White Gold Melted': whiteGold,
  'Chinese Silver Melted': chineseSilver,
  'Billon Melted': billon,
}

export const goldMaterial = {
  name: 'Gold',
  image: gold,
  price: 50,
  type: 'metal',
}

export const silverMaterial = {
  name: 'Silver',
  image: silver,
  price: 40,
  type: 'metal',
}

export const ironMaterial = {
  name: 'Iron',
  image: iron,
  price: 30,
  type: 'metal',
}

export const zincMaterial = {
  name: 'Zinc',
  image: zinc,
  price: 20,
  type: 'metal',
}

export const copperMaterial = {
  name: 'Copper',
  image: copper,
  price: 10,
  type: 'metal',
}

export const bronzeMaterial = {
  name: 'Bronze',
  image: bronze,
  price: 100,
  type: 'metal',
}

export const whiteGoldMaterial = {
  name: 'White Gold',
  image: whiteGold,
  price: 100,
  type: 'metal',
}

export const chineseSilverMaterial = {
  name: 'Chinese Silver',
  image: chineseSilver,
  price: 100,
  type: 'metal',
}

export const billonMaterial = {
  name: 'Billon',
  image: billon,
  price: 20,
  type: 'metal',
}

export const materials = [goldMaterial, silverMaterial, zincMaterial, ironMaterial, copperMaterial]
