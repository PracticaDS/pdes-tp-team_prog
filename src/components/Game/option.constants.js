import deleteOption from '../../assets/options/delete.png'
import rotateOption from '../../assets/options/rotate.png'
import moveOption from '../../assets/options/move.png'

export const editionOptions = [
  {
    image: deleteOption,
    title: 'Delete',
  },
  {
    image: rotateOption,
    title: 'Rotate',
  },
  {
    image: moveOption,
    title: 'Move',
  },
]
