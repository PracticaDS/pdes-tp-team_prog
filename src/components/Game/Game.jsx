import React from 'react'
import ResourcesBar from '../Resources/ResourcesBar'
import crafter from '../../assets/crafter.png'
import furnace from '../../assets/furnace.png'
import seller from '../../assets/seller.png'
import starter from '../../assets/starter.png'
import transporter from '../../assets/transporter.png'
import MachineTypes from '../MachineTypes/MachineTypes'
import EmptyMachineNode from '../MachineNodes/EmptyMachineNode/EmptyMachineNode'
import FactoryGrid from '../FactoryGrid/FactoryGrid'
import './Game.css'

const machineTypes = [
  {
    image: starter,
    title: 'Starter',
  },
  {
    image: seller,
    title: 'Seller',
  },
  {
    image: crafter,
    title: 'Crafter',
  },
  {
    image: furnace,
    title: 'Furnace',
  },
  {
    image: transporter,
    title: 'Transporter',
  },
]

const rowBlock = [
  EmptyMachineNode,
  EmptyMachineNode,
  EmptyMachineNode,
  EmptyMachineNode,
  EmptyMachineNode,
]
const blocks = [rowBlock, rowBlock, rowBlock, rowBlock, rowBlock]

// TODO: we dont need pass every state components
const Game = () => (
  <div className="Game">
    {/* When we have redux, we wont pass the currency */}
    <ResourcesBar currency={0} />
    <div className="HorizontalGameParts">
      <MachineTypes elements={machineTypes.slice(0, 5)} />
      <FactoryGrid componenets={blocks} />
      <MachineTypes elements={[]} /> {/* TODO: we need change this component */}
    </div>
  </div>
)

export default Game
