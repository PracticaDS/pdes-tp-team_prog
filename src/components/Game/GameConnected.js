import Game from './Game'
import GameConnector from './GameConnector'

export default GameConnector(Game)
