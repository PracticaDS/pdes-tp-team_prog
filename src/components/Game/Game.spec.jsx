import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Game from './Game'

it('render simple Game Component', () => {
  const state = {
    restartCurrency: () => true,
    startGame: () => true,
    tick: () => true,
    Grid: {
      dimensions: { n: 6, m: 6 },
      gridValues: Array(6).fill(Array(6).fill({ machine: { type: 'Empty' } })),
    },
    GameState: {
      currency: 1000,
    },
  }
  const mockStore = configureStore()
  const store = mockStore(state)
  const tree = renderer.create(
    <Provider store={store}>
      <Game />
    </Provider>,
  )
  expect(tree).not.toBe(null)
})
