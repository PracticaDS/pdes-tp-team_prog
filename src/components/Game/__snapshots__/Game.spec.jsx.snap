// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render simple Game Component 1`] = `
<div
  className="gameMainContainer"
>
  <div
    className="ResourcesBar"
  >
    <div
      className="Currency"
    >
      <font
        className="TitleFont"
      >
        Currency
      </font>
      <font
        className="CountFont"
      >
        $1000
      </font>
    </div>
  </div>
  <div
    className="HorizontalGameParts"
  >
    <div
      className="gamePanel gameLeftPanel"
    >
      <div
        className="mainLeftPanel"
      >
        <div>
          <font
            className="machineTypesHeader"
          >
            Machines
          </font>
          <div
            className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 machineTypesMainContainer"
          >
            <div
              className="MuiCardContent-root-29 machineTypesContainer"
            >
              <div
                className="machineTypesElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="machineElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="machineTypesElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="machineElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="machineTypesElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="machineElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="machineTypesElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="machineElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="machineTypesElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="machineElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "margin-top": "1rem",
            }
          }
        >
          <font
            className="editionHeader"
          >
            Edition
          </font>
          <div
            className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 editionMainContainer"
          >
            <div
              className="MuiCardContent-root-29 editionContainer"
            >
              <div
                className="editionElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="editionOptionElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="editionElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="editionOptionElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="editionElement"
              >
                <div
                  onClick={[Function]}
                >
                  <img
                    alt="myImage"
                    className="editionOptionElement"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="gameMidPanel"
    >
      <div
        className="FactoryGrid"
      >
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
        <div
          className="FactoryRow"
        >
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
          <div
            className="Block"
            onClick={[Function]}
          >
            <div
              className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="gamePanel gameRightPanel"
    >
      <div
        className="detail"
      >
        <h6
          className="MuiTypography-root-30 MuiTypography-h6-47 MuiTypography-gutterBottom-57 title text-content"
        >
          Details
        </h6>
        <div
          className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
        >
          <div
            className="MuiCardContent-root-29"
          >
            <p
              className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-57 text-content"
            >
              No Name
            </p>
            <p
              className="MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-gutterBottom-57 text-content"
            >
              Cost: 
              <span
                className="text-content-number"
              >
                $
                0
              </span>
            </p>
            <p
              className="MuiTypography-root-30 MuiTypography-body1-39 text-content"
            >
              Frequency: 
              <span
                className="text-content-number"
              >
                0
                /s
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
